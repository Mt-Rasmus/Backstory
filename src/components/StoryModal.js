
import React from 'react';
import PremiseView from './PremiseView';
import BackstoryView from './BackstoryView';
import Modal from 'react-modal';
if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

export default class StoryModal extends React.Component {
   constructor(props) {
      super(props);
      this.state = {
         solved: false,
         showBackstory: false
      };
   }

   toggleShowBackstory = () => {
      this.setState({showBackstory: !this.state.showBackstory})
   }

   resetModal = () => {
      this.props.handleCloseStoryModal();
      setTimeout(() => {this.setState({showBackstory: false})}, 500);
   }

   toggleInTransitionClass = () => {
      this.classList.toggle('rotate-div-in');
  }

   render () {
      return (
         <Modal
            isOpen={this.props.showModal}
            onRequestClose={this.resetModal} // clicking esc or outside modal
            closeTimeoutMS={200}      
            className="modal"
            style={{overlay:{backgroundColor:"rgba(0,0,0,0.3)"}}}
         >
            <span className="button--close" onClick={this.resetModal}></span>
            <div className="inner-modal">
               <div className="modal__title">
                  <img src={'/images/scratch_red1.png'} className="scratch-icon" alt=""/>
                  <p className="list-item__title">{this.props.story.title}</p>
               </div>
               {!this.state.showBackstory
                  ? <PremiseView story={this.props.story} toggleShowBackstory={this.toggleShowBackstory} />
                  : <BackstoryView story={this.props.story} toggleShowBackstory={this.toggleShowBackstory} />  
               }
               <div className="list-item__small-container">
                  <label className="list-item__small-text">Mark as solved:</label>
                  <input 
                     className="list-item__checkbox" 
                     type="checkbox" 
                     onClick={this.props.toggleSolved}
                     defaultChecked={this.props.solved}
                  />
               </div>
            </div>
         </Modal>
      )
   }   
}
