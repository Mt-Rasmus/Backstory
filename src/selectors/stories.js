
import database from '../firebase/firebase';

export const getStories = () => {
   return database.ref(`stories`).once("value").then((snapshot) => {
      return snapshot.val();
   })   
}

// localStorage.getItem(`story${this.props.story.id}_solved`)

export const getSolved = (solvedString) => {
   if (localStorage.getItem(solvedString) != null) {
      return localStorage.getItem(solvedString).then((snapshot) => {
         return snapshot.val();
      }) 
   } else {
      return false;
   }

}