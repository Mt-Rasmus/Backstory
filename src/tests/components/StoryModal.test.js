
import React from 'react';
import StoryModal from '../../components/StoryModal';
import { shallow } from 'enzyme';
import stories from '../../fixtures/teststories'; // importing test data

test('Should render StoryModal correctly', () => {
   const wrapper = shallow(<StoryModal story={stories[0]} />);
   expect(wrapper).toMatchSnapshot();
});
