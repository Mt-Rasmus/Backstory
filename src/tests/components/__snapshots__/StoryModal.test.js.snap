// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render StoryModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
      },
    }
  }
>
  <span
    className="button--close"
    onClick={[Function]}
  />
  <div
    className="inner-modal"
  >
    <div
      className="modal__title"
    >
      <img
        alt=""
        className="scratch-icon"
        src="/images/scratch_red1.png"
      />
      <p
        className="list-item__title"
      >
        Title of first story
      </p>
    </div>
    <PremiseView
      story={
        Object {
          "backstory": "The backstory",
          "id": "1",
          "premise": "A premise to a story",
          "title": "Title of first story",
        }
      }
      toggleShowBackstory={[Function]}
    />
    <div
      className="list-item__small-container"
    >
      <label
        className="list-item__small-text"
      >
        Mark as solved:
      </label>
      <input
        className="list-item__checkbox"
        type="checkbox"
      />
    </div>
  </div>
</Modal>
`;
